src/realm/object-store/sync/generic_network_transport.cpp

```diff
@@ -172,6 +172,23 @@ struct CustomErrorCategory : public std::error_category {
 
 CustomErrorCategory g_custom_error_category;
 
+static const std::map<std::string, ClientErrorCode> client_error_map = {
```

Comment:
It doesn't look like we actually use this as a map anywhere. Can you make this a `std::vector<std::pair<std::string, ClientErrorCode>>` instead? Also, you can turn this into a magic static by putting it inside the function where it's used like
```
const std::string& get_error_message(ClientErrorCode error) {
    static const std::vector<std::pair<std::string, ClientErrorCode>> client_errors = {
        {"user_not_found", ClientErrorCode::user_not_found},
...
    };
    if (auto it = std::find_if(client_errors.begin(), client_errors.end() [&](const auto& pair) {
        return pair.second == error;
    }); it != client_errors.end()) {
        return it->first;
    }
    return "unknown";
}
```